src/targets/gpu/driver/CMakeLists.txt

```diff
@@ -22,10 +22,22 @@
 # THE SOFTWARE.
 #####################################################################################
 
-file(GLOB GPU_DRIVER_SRCS ${CONFIGURE_DEPENDS} ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
-add_executable(gpu-driver
-    ${GPU_DRIVER_SRCS}
-)
-rocm_clang_tidy_check(gpu-driver)
-target_include_directories(gpu-driver PRIVATE include)
-target_link_libraries(gpu-driver PRIVATE migraphx_gpu)
+add_executable(migraphx_gpu_driver
+        action.cpp
+        include/migraphx/gpu/driver/action.hpp
+        compile_op.cpp
+        main.cpp
+        parser.cpp
+        include/migraphx/gpu/driver/parser.hpp
```

Comment:
instead of header files here can it make use of `target_include_directories` ?

Reply:
As of the specification, the CMake expects all files comprising an executable or library to be added to the source files list (even the ones not being compiled). In IDEs natively working with CMake, we use that information to view a complete project tree, symbol search, and more. For instance, JetBrains CLion, when a file is not added to the target's source files list, only `#included`, the tool is not considering that file in symbol searches (however, that can be overridden). Adding the `*.hpp` file to the list significantly speeds up project file operations (in contrast to parsing 1600 source files to build the dependency tree every couple of seconds when the content changes).

Reply:
However, the Windows will compile without adding *.hpp to the list.